app/assets/javascripts/forms.js.coffee

```diff
@@ -1,24 +1,60 @@
 $ ->
+  class Selectizer
+    constructor: (el) ->
+      @$el = $(el)
+
+    isFreeForm: ->
+      @$el.is('input')
+
+    src: ->
+      @$el.attr('data-src')
+
+    isRemote: ->
+      !!@src()
+
+    selectizeOpts: ->
+      opts = {}
+
+      if @isFreeForm()
+        opts.create = true
+        opts.maxItems = 1
+
+      if @isRemote()
+        attr = @$el.attr('data-attr')
+        opts.labelField = attr
+        opts.searchField = [attr]
+        opts.valueField = attr
+
+      opts
+
+    enable: ->
+      opts = @selectizeOpts()
+      @$el.selectize(opts)
+
+    selectizeObj: ->
+      @$el[0].selectize
+
+    onOptionsLoaded: (data) ->
+      selectize = @selectizeObj()
+      selectize.addOption(data)
+
+    loadRemoteOptions: ->
+      # TODO make sorting smarter, e.g. approvers/vendors they have used before
+      $.ajax(
+        url: @src()
+        data: {limit: 100}
+        dataType: 'json'
+        cache: true
+        context: @
+        success: @onOptionsLoaded
+      )
+
+    loadOptionsIfRemote: ->
+      if @isRemote()
+        @loadRemoteOptions()
+
+
   $('.js-selectize').each (i, el) ->
-    $el = $(el)
-    src = $el.attr('data-src')
-    opts = {}
-
-    if $el.is('input')
-      # allow free-form input
-      opts.create = true
-      opts.maxItems = 1
-
-    if src
-      # TODO make configurable
-      opts.labelField = 'email_address'
-      opts.searchField = ['email_address', 'first_name', 'last_name']
-      opts.valueField = 'email_address'
-
-    $el.selectize(opts)
-
-    if src
-      # load options from server
-      selectize = $el[0].selectize
-      $.getJSON src, (data) ->
-        selectize.addOption(data)
+    selectizer = new Selectizer(el)
+    selectizer.enable()
+    selectizer.loadOptionsIfRemote()
```

Comment:
I'm personally a fan of keeping constructors innocuous, but could go either way if you like it better.